cmd/bosun/expr/funcs.go

```diff
@@ -595,6 +603,56 @@ func Merge(e *State, T miniprofiler.Timer, series ...*Results) (*Results, error)
 	return res, nil
 }
 
+func LeftJoin(e *State, T miniprofiler.Timer, keysCSV, columnsCSV string, rowData ...*Results) (*Results, error) {
+	res := &Results{}
+	dataWidth := len(rowData)
+	if dataWidth == 0 {
+		return res, fmt.Errorf("leftjoin requires at least one item to populate rows")
+	}
+	keyColumns := strings.Split(keysCSV, ",")
+	dataColumns := strings.Split(columnsCSV, ",")
+	keyWidth := len(keyColumns)
+	keyIndex := make(map[string]int)
+	for i, v := range keyColumns {
+		keyIndex[v] = i
+	}
+	if len(dataColumns) != dataWidth {
+		return res, fmt.Errorf("mismatch in length of data rows and data labels")
+	}
+	t := Table{}
+	t.Columns = append(keyColumns, dataColumns...)
+	rowWidth := len(dataColumns) + len(keyColumns)
+	rowGroups := []opentsdb.TagSet{}
+	for i, r := range rowData {
+		if i == 0 {
+			for _, val := range r.Results {
+				row := make([]interface{}, rowWidth)
+				for k, v := range val.Group {
+					if ki, ok := keyIndex[k]; ok {
+						row[ki] = v
+					}
+				}
+				row[keyWidth+i] = val.Value.Value()
+				rowGroups = append(rowGroups, val.Group)
+				t.Rows = append(t.Rows, row)
+			}
+			continue
+		}
+		for rowIndex, group := range rowGroups {
+			for _, val := range r.Results {
+				if group.Subset(val.Group) {
+					t.Rows[rowIndex][keyWidth+i] = val.Value.Value()
```

Comment:
Since it is a left join, that can't happen as the column won't be joined. The requirement is that The columns are a subset of the first. So if your first column just has host, and the second has host,drive - it won't be joined because host,drive is not a subset of host. So if you do this you just end up with a column that has no values in it.